src/main/java/org/elasticsearch/search/aggregations/metrics/stats/extended/ExtendedStatsParser.java

```diff
@@ -18,22 +18,90 @@
 
+import org.elasticsearch.ElasticsearchParseException;
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.unit.RatioValue;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.search.SearchParseException;
+import org.elasticsearch.search.aggregations.Aggregator;
 import org.elasticsearch.search.aggregations.AggregatorFactory;
-import org.elasticsearch.search.aggregations.metrics.NumericValuesSourceMetricsAggregatorParser;
 import org.elasticsearch.search.aggregations.support.ValuesSource;
 import org.elasticsearch.search.aggregations.support.ValuesSourceConfig;
+import org.elasticsearch.search.aggregations.support.ValuesSourceParser;
+import org.elasticsearch.search.internal.SearchContext;
+
+import java.io.IOException;
 
 /**
  *
  */
-public class ExtendedStatsParser extends NumericValuesSourceMetricsAggregatorParser<InternalExtendedStats> {
+public class ExtendedStatsParser  implements Aggregator.Parser {
+
+    static final ParseField SIGMA = new ParseField("sigma");
+    static final ParseField CONFIDENCE_INTERVAL = new ParseField("confidence_interval");
+
+    @Override
+    public String type() {
+        return InternalExtendedStats.TYPE.name();
+    }
 
-    public ExtendedStatsParser() {
-        super(InternalExtendedStats.TYPE);
+    protected AggregatorFactory createFactory(String aggregationName, ValuesSourceConfig<ValuesSource.Numeric> config, double sigma, double confidenceLevel) {
+        return new ExtendedStatsAggregator.Factory(aggregationName, config, sigma, confidenceLevel);
     }
 
     @Override
-    protected AggregatorFactory createFactory(String aggregationName, ValuesSourceConfig<ValuesSource.Numeric> config) {
-        return new ExtendedStatsAggregator.Factory(aggregationName, config);
+    public AggregatorFactory parse(String aggregationName, XContentParser parser, SearchContext context) throws IOException {
+
+        ValuesSourceParser<ValuesSource.Numeric> vsParser = ValuesSourceParser.numeric(aggregationName, InternalExtendedStats.TYPE, context).formattable(true)
+                .build();
+
+        XContentParser.Token token;
+        String currentFieldName = null;
+        double sigma = 2.0;
+        double confidenceInterval = -1;
+
+        while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
+            if (token == XContentParser.Token.FIELD_NAME) {
+                currentFieldName = parser.currentName();
+            } else if (vsParser.token(currentFieldName, token, parser)) {
+                continue;
+            } else if (token == XContentParser.Token.VALUE_NUMBER) {
+                if (SIGMA.match(currentFieldName)) {
+                    sigma = parser.intValue();
+                } else if (CONFIDENCE_INTERVAL.match(currentFieldName)) {
+                    confidenceInterval = parser.doubleValue();
+                    if (confidenceInterval < 0 || confidenceInterval > 0.99994) {
```

Comment:
The confidence interval is worked out using a Z-table. The formula for working out the value of Z is a complex integral with no mathematical solution so the solutions are numerical. We have a Z-Table with values for confidence from 0 to 0.99994, so it just so happens to be where our Z-Table stops. Since the confidence interval approaches 1 asymptotically, we have to stop somewhere before 1 and this value seems like it will cover almost all use-cases so should be adequate. Does that make sense?